### ISSUE TITLE
memory leak using crypto sha256 hashing

### ISSUE THREAD

**Issue by add1ed** [oven-sh/bun#3250]
### What version of Bun is running?

0.6.7

### What platform is your computer?

Darwin 22.4.0 arm64 arm

### What steps can reproduce the bug?

Run the following code:
```
const crypto = require('crypto');

function sha256(buf) {
  return crypto.createHash('sha256').update(buf).digest();
};

async function main() {
  for (var i = 1000000; i >= 0; i--) {
    const buf = Buffer.alloc(2046);
    const hash = sha256(buf);
    if (i % 1000 === 0) {
      await new Promise(r => setTimeout(r, 20));
      global.gc ? global.gc() : Bun?.gc(true);
    }
  }
}

main();
```
using `bun index.js` and watch the memory usage, as reported by the OS (in my case looking at Activity Monitor).

(And repeat using `node --expose-gc index.js` to observe the equivalent behaviour in nodejs.)

### What is the expected behavior?

Memory usage stays flat.

### What do you see instead?

Memory usage grows.

### Additional information

Outputting `heapStats()` data from `bun:jsc` at various points during the looping indicates heap memory stays flat.

Memory seems to approx grow at ~130 bytes per hash call.

I have observed, in my own application, that over the course of an hour, in which ~7.2M hashes are performed, memory usage grows by ~1GB.

**Comment by Jarred-Sumner** [comment]
Thank you for the report.

After:
<img width="451" alt="image" src="https://github.com/oven-sh/bun/assets/709451/d111274a-0a4a-4075-82f2-39699dbf8521">

Before:
<img width="444" alt="image" src="https://github.com/oven-sh/bun/assets/709451/231f7e38-0717-4610-a8dd-aebac2aa70d7">

Node.js, for comparison:
<img width="454" alt="image" src="https://github.com/oven-sh/bun/assets/709451/22e4466e-883e-4659-9f5c-f0310865515b">

**Comment by BinaryFox83** [comment]
> Thank you for the report.
> 
> After: <img alt="image" width="451" src="https://user-images.githubusercontent.com/709451/244854940-d111274a-0a4a-4075-82f2-39699dbf8521.png">
> 
> Before: <img alt="image" width="444" src="https://user-images.githubusercontent.com/709451/244854950-231f7e38-0717-4610-a8dd-aebac2aa70d7.png">
> 
> Node.js, for comparison: <img alt="image" width="454" src="https://user-images.githubusercontent.com/709451/244854952-22e4466e-883e-4659-9f5c-f0310865515b.png">

Is the fix already present in 0.6.8 or we have to wait for the next one?
